tests/test_PsfFlux.py

```diff
@@ -144,11 +180,19 @@ def testMonteCarlo(self):
                 algorithm.measure(record, exposure)
                 instFluxes.append(record.get("base_PsfFlux_instFlux"))
                 instFluxErrs.append(record.get("base_PsfFlux_instFluxErr"))
+                measuredChi2s.append(record.get("base_PsfFlux_chi2"))
+                measuredNPixels.append(record.get("base_PsfFlux_npixels"))
+                chi2, nPixels = compute_chi2(exposure, record.getCentroid(), instFluxes[-1])
+                expectedChi2s.append(chi2)
+                expectedNPixels = nPixels  # should be the same each time
             instFluxMean = np.mean(instFluxes)
             instFluxErrMean = np.mean(instFluxErrs)
             instFluxStandardDeviation = np.std(instFluxes)
             self.assertFloatsAlmostEqual(instFluxErrMean, instFluxStandardDeviation, rtol=0.10)
             self.assertLess(abs(instFluxMean - instFlux), 2.0*instFluxErrMean / nSamples**0.5)
+            self.assertFloatsAlmostEqual(np.mean(measuredChi2s), np.mean(expectedChi2s), rtol=1e-7)
```

Comment:
I don't understand why you're comparing the means here. Shouldn't `measuredChi2s[i] == expectedChi2s[i]` for all `i`?

Reply:
No, that's only true for the number of pixels. The chi2 is different for each individual sample at a given noise level because the data+noise is randomly generated.

Reply:
Oh! I get your comment now. That's absolutely right: I've changed it to do the comparison per-sample.